cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -3865,6 +3905,221 @@ if(ARROW_ORC)
+    # OpenTelemetry tries to determine the processor arch for vcpkg, which fails
+    # on s390x, even though it doesn't use vcpkg there. Tell it ARCH manually
+    externalproject_add(opentelemetry_ep
+                        ${EP_LOG_OPTIONS}
+                        URL_HASH "SHA256=${ARROW_OPENTELEMETRY_BUILD_SHA256_CHECKSUM}"
+                        LIST_SEPARATOR ${OPENTELEMETRY_PREFIX_PATH_LIST_SEP_CHAR}
+                        CONFIGURE_COMMAND ${CMAKE_COMMAND} -E env ARCH=s390x
+                                          ${CMAKE_COMMAND} -G ${CMAKE_GENERATOR}
+                                          "<SOURCE_DIR><SOURCE_SUBDIR>"
+                                          ${OPENTELEMETRY_CMAKE_ARGS}
+                        BUILD_COMMAND ${CMAKE_COMMAND} --build "<BINARY_DIR>" --target all
+                        INSTALL_COMMAND ${CMAKE_COMMAND} --build "<BINARY_DIR>" --target
+                                        install
+                        URL ${OPENTELEMETRY_SOURCE_URL}
+                        BUILD_BYPRODUCTS ${OPENTELEMETRY_BUILD_BYPRODUCTS}
+                        EXCLUDE_FROM_ALL NOT
+                        ${ARROW_WITH_OPENTELEMETRY}
+                        DEPENDS ${_OPENTELEMETRY_DEPENDENCIES})
+  else()
+    externalproject_add(opentelemetry_ep
+                        ${EP_LOG_OPTIONS}
+                        URL_HASH "SHA256=${ARROW_OPENTELEMETRY_BUILD_SHA256_CHECKSUM}"
+                        LIST_SEPARATOR ${OPENTELEMETRY_PREFIX_PATH_LIST_SEP_CHAR}
+                        CMAKE_ARGS ${OPENTELEMETRY_CMAKE_ARGS}
+                        URL ${OPENTELEMETRY_SOURCE_URL}
+                        BUILD_BYPRODUCTS ${OPENTELEMETRY_BUILD_BYPRODUCTS}
+                        EXCLUDE_FROM_ALL NOT
+                        ${ARROW_WITH_OPENTELEMETRY}
+                        DEPENDS ${_OPENTELEMETRY_DEPENDENCIES})
+  endif()
+
+  if(ARROW_WITH_OPENTELEMETRY)
+    externalproject_add_step(opentelemetry_ep download_proto
+                             COMMAND ${CMAKE_COMMAND} -E copy_directory
+                                     $<TARGET_PROPERTY:opentelemetry_proto_ep,_EP_SOURCE_DIR>/opentelemetry
+                                     $<TARGET_PROPERTY:opentelemetry_ep,_EP_SOURCE_DIR>/third_party/opentelemetry-proto/opentelemetry
+                             DEPENDEES download
+                             DEPENDERS configure)
+  endif()
+
+  add_dependencies(toolchain opentelemetry_ep)
+  add_dependencies(toolchain-tests opentelemetry_ep)
+
+  set(OPENTELEMETRY_VENDORED 1)
+
+  if(NOT ${CMAKE_VERSION} VERSION_LESS "3.15.0")
+    # Make sure all installed artifacts get cleaned up. Important for the RPM
+    # builds which fail if anything is left behind. Things in BUILD_BYPRODUCTS
+    # get cleaned automatically, but we can't list every header installed.
+    set_property(TARGET opentelemetry_ep
+                 APPEND
+                 PROPERTY ADDITIONAL_CLEAN_FILES "${OPENTELEMETRY_PREFIX}")
+  endif()
```

Comment:
I think that we don't need this.
If we don't run any steps of `opentelemetry_ep` on `ninja install`, the RPM related problem can be solved. The `ninja install` in the RPM build is ran with `DESTDIR`. If the install step of `opentelemetry_ep` is ran with `ninja install`, OpenTelemetry related files are installed into `${DESTDIR}/....`. It causes the RPM related problem.

Reply:
Ah, sorry, I left this behind. I've removed it.

It no longer runs any `opentelemetry_ep` steps in `ninja install`, though what I observed was just it generating stamp files, not actually running the install step.